spring-batch-infrastructure/src/main/java/org/springframework/batch/support/transaction/TransactionAwareBufferedWriter.java

```diff
@@ -71,6 +75,47 @@ public TransactionAwareBufferedWriter(FileChannel channel, Runnable closeCallbac
 	public void setEncoding(String encoding) {
 		this.encoding = encoding;
 	}
+	
+	/**
+	 * Buffers the output on a {@code byte[]} level. Compared to
+	 * buffering on a {@code char[]} level this saves about 50% memory
+	 * for common encodings and input.
+	 */
+	static final class Buffer {
```

Comment:
Do we really need this new `Buffer` class? Can't we use `java.nio.ByteBuffer` or probably `org.springframework.core.io.buffer.DataBuffer`? I'm not sure if this is possible, I'm just trying to see if we can reuse an existing data strucuture before creating a new one.

Reply:
I don't think so. `java.nio.ByteBuffer` has a fixed maximum size (capacity) and can not grow beyond this. We don't really know how much we have to buffer, so we would have to allocate new ones once we exceed their capacity.
`org.springframework.core.io.buffer.DefaultDataBuffer` is just wrapper around a `java.nio.ByteBuffer` and ends up inheriting its limitations.